### ISSUE TITLE
tee_source_ip no longer works as of ver >= 1.7.6

### ISSUE THREAD

**Issue by floatingstatic** [pmacct/pmacct#493]
**Description**
It appears that as of [this commit](https://github.com/pmacct/pmacct/commit/a35e3c4d4f5b4f96cc008a4ad8004f3ed04fab11) the `tee_source_ip` configuration no longer works. I am using the tee plugin with nfacctd. My nfacctd.conf is very simple (ip addresses sanitized from their actual values):

```
plugins: tee
tee_receivers: /persist/sys/pmacct/receivers
tee_source_ip: 203.0.113.0
```

`/persist/sys/pmacct/receivers` looks something like this:

```
!
! tee_receivers: Tee receivers map
!

! ips needs to be stringified in the form `ip:port,ip:port,` (dangling coma allowed)
! balance-alg=rr will load balance betweeen the destinations. no effect with 1 ip

!
id=1	ip=203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,203.0.113.254:4739,  balance-alg=rr
```

Note in the above the same receiver IP is used multiple times just to generate hash entropy for my load balancing application (not really relevant to the problem).

I observe this in nfacctd running in debug mode:

```
DEBUG ( default/core ): Received NetFlow/IPFIX packet from [203.0.113.0:58849] version [10] seqno [2185176]
DEBUG ( default/core ): Received NetFlow/IPFIX packet from [203.0.113.0:58849] version [10] seqno [2185182]
DEBUG ( default_tee/tee ): Sending NetFlow/IPFIX packet from [203.0.113.0:58849] seqno [2185158] to [203.0.113.254:4739]
DEBUG ( default_tee/tee ): Sending NetFlow/IPFIX packet from [203.0.113.0:58849] seqno [2185164] to [203.0.113.254:4739]
DEBUG ( default_tee/tee ): Sending NetFlow/IPFIX packet from [203.0.113.0:58849] seqno [2185170] to [203.0.113.254:4739]
DEBUG ( default_tee/tee ): Sending NetFlow/IPFIX packet from [203.0.113.0:58849] seqno [2185176] to [203.0.113.254:4739]
DEBUG ( default_tee/tee ): Sending NetFlow/IPFIX packet from [203.0.113.0:58849] seqno [2185182] to [203.0.113.254:4739]
```

On the wire however I see that the source IP is not being overwritten:

```
12:44:54.764824 Out 74:83:ef:1e:89:0d (oui Unknown) ethertype IPv4 (0x0800), length 516: 172.24.12.1.34162 > 203.0.113.254.ipfix: UDP, length 472
12:44:54.764851 Out 74:83:ef:1e:89:0d (oui Unknown) ethertype IPv4 (0x0800), length 516: 172.24.12.1.54731 > 203.0.113.254.ipfix: UDP, length 472
12:44:54.764887 Out 74:83:ef:1e:89:0d (oui Unknown) ethertype IPv4 (0x0800), length 516: 172.24.12.1.41841 > 203.0.113.254.ipfix: UDP, length 472
12:44:54.764915 Out 74:83:ef:1e:89:0d (oui Unknown) ethertype IPv4 (0x0800), length 516: 172.24.12.1.37585 > 203.0.113.254.ipfix: UDP, length 472
12:44:54.764940 Out 74:83:ef:1e:89:0d (oui Unknown) ethertype IPv4 (0x0800), length 364: 172.24.12.1.50440 > 203.0.113.254.ipfix: UDP, length 320
```

As best as I can tell `config.nfprobe_source_ip` is no longer being referenced in the code as of the commit referenced above. I would re-add it as it was previously but it is not clear to me if doing so will have a negative affect on the transparent mode feature.

**Version**
I'm currently running from a recent commit to `master`

```
$ nfacctd -V
NetFlow Accounting Daemon, nfacctd 1.7.7-git [20210408-2 (9539b81)]
```

**Appreciation**
Please consider starring this project to boost our reach on github!

**Comment by paololucente** [comment]
Hi Jeremiah,

I have closed #495 and should have re-worked the patch so that the function keeps being reusable elsewhere in the code. Can you give it a try to see if it works for you?

Paolo

**Comment by floatingstatic** [comment]
@paololucente I pulled in the latest changes and can confirm this has fixed the issue. Thanks again!
